### ISSUE TITLE
Update pgadmin to v4

### ISSUE THREAD

**Issue by srghma** [NixOS/nixpkgs#33103]
Please update pgadmin

**Comment by emmanuelrosa** [comment]
Having looked into this, I determined the following:

1. `pgAdmin4` is a completely different product than `pgAdmin3`. As such, it is not a mere update.
2. `pgAdmin4` is fundamentally a web application written in Python/Flash, SQLite, and HTML/JavaScript/CSS.
3. There's a *Desktop* variant of the app which is essentially the web app with a Qt-based wrapper.

Given the above, within the context of Nix/NixOS this means:

1. The Nix expression(s) for the package would need to use the Nix Packages Collection's programming-language-specific tooling for Python and Qt, in order to build the application.
2. A NixOS module would need to be created to allow the use of the *server* variant of the application. Being a WSGI application, a web server also needs to be configured to act as a reverse proxy.
3. The web application relies on having access to directories (ex. `/var`) in order to save app state; This must be considered when creating the NixOS module.
4. I'm probably overlooking something.

I mention all this merely to provide some indication as to the level of complexity involved in adding `pgAdmin4`.

**Comment by BjornMelgaard** [comment]
@emmanuelrosa just make pip2nix like expression
I'm using pgadmin4 like this `docker run --rm -p 5050:5050 thajeztah/pgadmin4` now and living fine

**Comment by BjornMelgaard** [comment]
though its not saving application state

**Comment by emmanuelrosa** [comment]
If `pip2nix` is all you need, I say go for it. With so many things to do in Nix land and so few hands on deck, your contribution would be appreciated.

**Comment by steve-chavez** [comment]
still important to me

**Comment by costrouc** [comment]
This issue is important to me as well. Is there an easy way to hack this by using an existing RPM? I know this would be a dirty solution but would about the javascript dependency problem. https://www.pgadmin.org/download/pgadmin-4-rpm/

**Comment by steve-chavez** [comment]
Giving this a shot. Does anyone know what package contains the `krb5-config` command?

I've installed `libkrb5`, `krb5`, `krb5Full`, `kerberos`, but no dice.

**Comment by evanjs** [comment]
> Giving this a shot. Does anyone know what package contains the `krb5-config` command?
> 
> I've installed `libkrb5`, `krb5`, `krb5Full`, `kerberos`, but no dice.
> 
> Building pgadmin4 requires `krb5-config`. I'm getting this error:
> 
> ```
> subprocess.CalledProcessError: Command 'krb5-config --libs gssapi' returned non-zero exit status 127.
> ```

`nix-locate -w1 krb5-config` returns:
```
kerberos.dev
krb5.dev
heimdal.dev
```

[`nix-index`](https://github.com/bennofs/nix-index) is extremely helpful for this sort of thing.
___
`krb5` seems to work fine, and I get seemingly valid output from ` krb5-config --libs gssapi`:

```
-L/nix/store/8swdycf22zx7jhl5jmw7iczk75875xf7-krb5-1.18/lib -Wl,--enable-new-dtags -Wl,-rpath -Wl,/nix/store/8swdycf22zx7jhl5jmw7iczk75875xf7-krb5-1.18/lib -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err
```

**Comment by steve-chavez** [comment]
Thanks a lot! @evanjs.

Can confirm that `kerb5.dev` works :ok_hand: .

> nix-index is extremely helpful for this sort of thing

Awesome, I'll add nix-index to my toolbelt.

**Comment by mkg20001** [comment]
I'm currently working on the package for pgadmin4 https://github.com/NixOS/nixpkgs/pull/128986

**Comment by 1000101** [comment]
#128986 is not being worked on anymore

**Comment by ajs124** [comment]
Closed by #154764 and  #161521
